packages/flutter_tools/test/general.shard/version_test.dart

```diff
@@ -331,12 +331,18 @@ void main() {
       const String flutterNonStandardUrlDotGit = 'https://githubmirror.com/flutter/flutter.git';
       const String flutterStandardSshUrl = 'git@github.com:flutter/flutter';
 
+      FakePlatform createTestPlatform({String flutterGitUrl}){
+        return FakePlatform(
+          environment: <String,String> {'FLUTTER_GIT_URL': flutterGitUrl},
+        );
+      }
+
       testUsingContext('throws if repository url is null', () async {
         final FlutterVersion flutterVersion = FakeFlutterVersion(
           // repositoryUrl is null by default
         );
         try {
-          VersionUpstreamValidator(version: flutterVersion).run();
+          VersionUpstreamValidator(version: flutterVersion, platform: FakePlatform()).run();
```

Comment:
I can see that before we were injecting the platform, we were accidentally leaking the real system platform into the test suite, which is a huge cause of flakiness in our tool test suite. For the tests that only depend on a created instance of `VersionUpstreamValidator`, can you use `testWithoutContext()` instead of `testUsingContext()`? With `testWithoutContext()`, any test that tries to call `context.get()` via the `globals.` or any other method will throw. This way we know exactly what our the code depends on, and we don't get tests accidentally passing because the system they were running on had the correct platform setup. I imagine you will have to also inject `flutterGitSshUrl` via the constructor.

Reply:
We would also need to provide `globals.flutterGit` too if that's the case.

Reply:
...which adds two parameters which similar semantics?
```dart
VersionUpstreamValidator({
  required this.version,
  required this.platform,
  required this.flutterGitUrl,
  required this.flutterGitSshUrl,
});
```
Can't think of a better name to give to these parameters.

Reply:
Yeah, that feels bad. However, the only other way I can think of would be to define a class for `GitHubRemote`, that takes in an org and project name, and has two getters for `httpsUrl` and `sshUrl`, but that seems like overkill?